// @passageidentity/passage-node/lib/esm/generated/apis/MagicLinksApi.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import * as runtime from '../runtime';
import { CreateMagicLinkRequestToJSON, MagicLinkResponseFromJSON, } from '../models/index';
/**
 *
 */
export class MagicLinksApi extends runtime.BaseAPI {
    /**
     * Create magic link for a user.
     * Create Embeddable Magic Link
     */
    createMagicLinkRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters.appId === null || requestParameters.appId === undefined) {
                throw new runtime.RequiredError('appId', 'Required parameter requestParameters.appId was null or undefined when calling createMagicLink.');
            }
            if (requestParameters.createMagicLinkRequest === null || requestParameters.createMagicLinkRequest === undefined) {
                throw new runtime.RequiredError('createMagicLinkRequest', 'Required parameter requestParameters.createMagicLinkRequest was null or undefined when calling createMagicLink.');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("bearerAuth", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/apps/{app_id}/magic-links`.replace(`{${"app_id"}}`, encodeURIComponent(String(requestParameters.appId))),
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: CreateMagicLinkRequestToJSON(requestParameters.createMagicLinkRequest),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => MagicLinkResponseFromJSON(jsonValue));
        });
    }
    /**
     * Create magic link for a user.
     * Create Embeddable Magic Link
     */
    createMagicLink(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.createMagicLinkRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
}
