// @passageidentity/passage-node/lib/cjs/generated/models/AppResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppResponseToJSON = exports.AppResponseFromJSONTyped = exports.AppResponseFromJSON = exports.instanceOfAppResponse = void 0;
const AppInfo_1 = require("./AppInfo");
/**
 * Check if a given object implements the AppResponse interface.
 */
function instanceOfAppResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "app" in value;
    return isInstance;
}
exports.instanceOfAppResponse = instanceOfAppResponse;
function AppResponseFromJSON(json) {
    return AppResponseFromJSONTyped(json, false);
}
exports.AppResponseFromJSON = AppResponseFromJSON;
function AppResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'app': (0, AppInfo_1.AppInfoFromJSON)(json['app']),
    };
}
exports.AppResponseFromJSONTyped = AppResponseFromJSONTyped;
function AppResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'app': (0, AppInfo_1.AppInfoToJSON)(value.app),
    };
}
exports.AppResponseToJSON = AppResponseToJSON;
