// @passageidentity/passage-node/lib/cjs/generated/models/CreateUserRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUserRequestToJSON = exports.CreateUserRequestFromJSONTyped = exports.CreateUserRequestFromJSON = exports.instanceOfCreateUserRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CreateUserRequest interface.
 */
function instanceOfCreateUserRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "phone" in value;
    return isInstance;
}
exports.instanceOfCreateUserRequest = instanceOfCreateUserRequest;
function CreateUserRequestFromJSON(json) {
    return CreateUserRequestFromJSONTyped(json, false);
}
exports.CreateUserRequestFromJSON = CreateUserRequestFromJSON;
function CreateUserRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'email': json['email'],
        'phone': json['phone'],
        'user_metadata': !(0, runtime_1.exists)(json, 'user_metadata') ? undefined : json['user_metadata'],
    };
}
exports.CreateUserRequestFromJSONTyped = CreateUserRequestFromJSONTyped;
function CreateUserRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
        'phone': value.phone,
        'user_metadata': value.user_metadata,
    };
}
exports.CreateUserRequestToJSON = CreateUserRequestToJSON;
