// @passageidentity/passage-node/lib/cjs/generated/models/ElementCustomization.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElementCustomizationToJSON = exports.ElementCustomizationFromJSONTyped = exports.ElementCustomizationFromJSON = exports.instanceOfElementCustomization = void 0;
const runtime_1 = require("../runtime");
const FontFamily_1 = require("./FontFamily");
/**
 * Check if a given object implements the ElementCustomization interface.
 */
function instanceOfElementCustomization(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfElementCustomization = instanceOfElementCustomization;
function ElementCustomizationFromJSON(json) {
    return ElementCustomizationFromJSONTyped(json, false);
}
exports.ElementCustomizationFromJSON = ElementCustomizationFromJSON;
function ElementCustomizationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'passage_container_background_color': !(0, runtime_1.exists)(json, 'passage_container_background_color') ? undefined : json['passage_container_background_color'],
        'passage_container_max_width': !(0, runtime_1.exists)(json, 'passage_container_max_width') ? undefined : json['passage_container_max_width'],
        'passage_input_box_background_color': !(0, runtime_1.exists)(json, 'passage_input_box_background_color') ? undefined : json['passage_input_box_background_color'],
        'passage_input_box_border_radius': !(0, runtime_1.exists)(json, 'passage_input_box_border_radius') ? undefined : json['passage_input_box_border_radius'],
        'passage_header_font_family': !(0, runtime_1.exists)(json, 'passage_header_font_family') ? undefined : (0, FontFamily_1.FontFamilyFromJSON)(json['passage_header_font_family']),
        'passage_body_font_family': !(0, runtime_1.exists)(json, 'passage_body_font_family') ? undefined : (0, FontFamily_1.FontFamilyFromJSON)(json['passage_body_font_family']),
        'passage_header_text_color': !(0, runtime_1.exists)(json, 'passage_header_text_color') ? undefined : json['passage_header_text_color'],
        'passage_body_text_color': !(0, runtime_1.exists)(json, 'passage_body_text_color') ? undefined : json['passage_body_text_color'],
        'passage_primary_button_background_color': !(0, runtime_1.exists)(json, 'passage_primary_button_background_color') ? undefined : json['passage_primary_button_background_color'],
        'passage_primary_button_text_color': !(0, runtime_1.exists)(json, 'passage_primary_button_text_color') ? undefined : json['passage_primary_button_text_color'],
        'passage_primary_button_hover_color': !(0, runtime_1.exists)(json, 'passage_primary_button_hover_color') ? undefined : json['passage_primary_button_hover_color'],
        'passage_primary_button_border_radius': !(0, runtime_1.exists)(json, 'passage_primary_button_border_radius') ? undefined : json['passage_primary_button_border_radius'],
        'passage_primary_button_border_color': !(0, runtime_1.exists)(json, 'passage_primary_button_border_color') ? undefined : json['passage_primary_button_border_color'],
        'passage_primary_button_border_width': !(0, runtime_1.exists)(json, 'passage_primary_button_border_width') ? undefined : json['passage_primary_button_border_width'],
        'passage_secondary_button_background_color': !(0, runtime_1.exists)(json, 'passage_secondary_button_background_color') ? undefined : json['passage_secondary_button_background_color'],
        'passage_secondary_button_text_color': !(0, runtime_1.exists)(json, 'passage_secondary_button_text_color') ? undefined : json['passage_secondary_button_text_color'],
        'passage_secondary_button_hover_color': !(0, runtime_1.exists)(json, 'passage_secondary_button_hover_color') ? undefined : json['passage_secondary_button_hover_color'],
        'passage_secondary_button_border_radius': !(0, runtime_1.exists)(json, 'passage_secondary_button_border_radius') ? undefined : json['passage_secondary_button_border_radius'],
        'passage_secondary_button_border_color': !(0, runtime_1.exists)(json, 'passage_secondary_button_border_color') ? undefined : json['passage_secondary_button_border_color'],
        'passage_secondary_button_border_width': !(0, runtime_1.exists)(json, 'passage_secondary_button_border_width') ? undefined : json['passage_secondary_button_border_width'],
    };
}
exports.ElementCustomizationFromJSONTyped = ElementCustomizationFromJSONTyped;
function ElementCustomizationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'passage_container_background_color': value.passage_container_background_color,
        'passage_container_max_width': value.passage_container_max_width,
        'passage_input_box_background_color': value.passage_input_box_background_color,
        'passage_input_box_border_radius': value.passage_input_box_border_radius,
        'passage_header_font_family': (0, FontFamily_1.FontFamilyToJSON)(value.passage_header_font_family),
        'passage_body_font_family': (0, FontFamily_1.FontFamilyToJSON)(value.passage_body_font_family),
        'passage_header_text_color': value.passage_header_text_color,
        'passage_body_text_color': value.passage_body_text_color,
        'passage_primary_button_background_color': value.passage_primary_button_background_color,
        'passage_primary_button_text_color': value.passage_primary_button_text_color,
        'passage_primary_button_hover_color': value.passage_primary_button_hover_color,
        'passage_primary_button_border_radius': value.passage_primary_button_border_radius,
        'passage_primary_button_border_color': value.passage_primary_button_border_color,
        'passage_primary_button_border_width': value.passage_primary_button_border_width,
        'passage_secondary_button_background_color': value.passage_secondary_button_background_color,
        'passage_secondary_button_text_color': value.passage_secondary_button_text_color,
        'passage_secondary_button_hover_color': value.passage_secondary_button_hover_color,
        'passage_secondary_button_border_radius': value.passage_secondary_button_border_radius,
        'passage_secondary_button_border_color': value.passage_secondary_button_border_color,
        'passage_secondary_button_border_width': value.passage_secondary_button_border_width,
    };
}
exports.ElementCustomizationToJSON = ElementCustomizationToJSON;
