// @passageidentity/passage-node/lib/cjs/generated/models/Layouts.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LayoutsToJSON = exports.LayoutsFromJSONTyped = exports.LayoutsFromJSON = exports.instanceOfLayouts = void 0;
const LayoutConfig_1 = require("./LayoutConfig");
/**
 * Check if a given object implements the Layouts interface.
 */
function instanceOfLayouts(value) {
    let isInstance = true;
    isInstance = isInstance && "profile" in value;
    isInstance = isInstance && "registration" in value;
    return isInstance;
}
exports.instanceOfLayouts = instanceOfLayouts;
function LayoutsFromJSON(json) {
    return LayoutsFromJSONTyped(json, false);
}
exports.LayoutsFromJSON = LayoutsFromJSON;
function LayoutsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'profile': (json['profile'].map(LayoutConfig_1.LayoutConfigFromJSON)),
        'registration': (json['registration'].map(LayoutConfig_1.LayoutConfigFromJSON)),
    };
}
exports.LayoutsFromJSONTyped = LayoutsFromJSONTyped;
function LayoutsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'profile': (value.profile.map(LayoutConfig_1.LayoutConfigToJSON)),
        'registration': (value.registration.map(LayoutConfig_1.LayoutConfigToJSON)),
    };
}
exports.LayoutsToJSON = LayoutsToJSON;
