// @passageidentity/passage-node/lib/cjs/generated/models/ListDevicesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDevicesResponseToJSON = exports.ListDevicesResponseFromJSONTyped = exports.ListDevicesResponseFromJSON = exports.instanceOfListDevicesResponse = void 0;
const WebAuthnDevices_1 = require("./WebAuthnDevices");
/**
 * Check if a given object implements the ListDevicesResponse interface.
 */
function instanceOfListDevicesResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "devices" in value;
    return isInstance;
}
exports.instanceOfListDevicesResponse = instanceOfListDevicesResponse;
function ListDevicesResponseFromJSON(json) {
    return ListDevicesResponseFromJSONTyped(json, false);
}
exports.ListDevicesResponseFromJSON = ListDevicesResponseFromJSON;
function ListDevicesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'devices': (json['devices'].map(WebAuthnDevices_1.WebAuthnDevicesFromJSON)),
    };
}
exports.ListDevicesResponseFromJSONTyped = ListDevicesResponseFromJSONTyped;
function ListDevicesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'devices': (value.devices.map(WebAuthnDevices_1.WebAuthnDevicesToJSON)),
    };
}
exports.ListDevicesResponseToJSON = ListDevicesResponseToJSON;
