// @passageidentity/passage-node/lib/cjs/generated/models/MagicLinkType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicLinkTypeToJSON = exports.MagicLinkTypeFromJSONTyped = exports.MagicLinkTypeFromJSON = exports.MagicLinkType = void 0;
/**
 *
 * @export
 */
exports.MagicLinkType = {
    Login: 'login',
    VerifyIdentifier: 'verify_identifier'
};
function MagicLinkTypeFromJSON(json) {
    return MagicLinkTypeFromJSONTyped(json, false);
}
exports.MagicLinkTypeFromJSON = MagicLinkTypeFromJSON;
function MagicLinkTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.MagicLinkTypeFromJSONTyped = MagicLinkTypeFromJSONTyped;
function MagicLinkTypeToJSON(value) {
    return value;
}
exports.MagicLinkTypeToJSON = MagicLinkTypeToJSON;
