// @passageidentity/passage-node/lib/cjs/generated/models/MagicLinkResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicLinkResponseToJSON = exports.MagicLinkResponseFromJSONTyped = exports.MagicLinkResponseFromJSON = exports.instanceOfMagicLinkResponse = void 0;
const MagicLink_1 = require("./MagicLink");
/**
 * Check if a given object implements the MagicLinkResponse interface.
 */
function instanceOfMagicLinkResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "magic_link" in value;
    return isInstance;
}
exports.instanceOfMagicLinkResponse = instanceOfMagicLinkResponse;
function MagicLinkResponseFromJSON(json) {
    return MagicLinkResponseFromJSONTyped(json, false);
}
exports.MagicLinkResponseFromJSON = MagicLinkResponseFromJSON;
function MagicLinkResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'magic_link': (0, MagicLink_1.MagicLinkFromJSON)(json['magic_link']),
    };
}
exports.MagicLinkResponseFromJSONTyped = MagicLinkResponseFromJSONTyped;
function MagicLinkResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'magic_link': (0, MagicLink_1.MagicLinkToJSON)(value.magic_link),
    };
}
exports.MagicLinkResponseToJSON = MagicLinkResponseToJSON;
