// @passageidentity/passage-node/lib/cjs/generated/models/Model500Error.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Model500ErrorToJSON = exports.Model500ErrorFromJSONTyped = exports.Model500ErrorFromJSON = exports.instanceOfModel500Error = exports.Model500ErrorCodeEnum = void 0;
/**
 * @export
 */
exports.Model500ErrorCodeEnum = {
    InternalServerError: 'internal_server_error'
};
/**
 * Check if a given object implements the Model500Error interface.
 */
function instanceOfModel500Error(value) {
    let isInstance = true;
    isInstance = isInstance && "code" in value;
    isInstance = isInstance && "error" in value;
    return isInstance;
}
exports.instanceOfModel500Error = instanceOfModel500Error;
function Model500ErrorFromJSON(json) {
    return Model500ErrorFromJSONTyped(json, false);
}
exports.Model500ErrorFromJSON = Model500ErrorFromJSON;
function Model500ErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': json['code'],
        'error': json['error'],
    };
}
exports.Model500ErrorFromJSONTyped = Model500ErrorFromJSONTyped;
function Model500ErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'error': value.error,
    };
}
exports.Model500ErrorToJSON = Model500ErrorToJSON;
