// @passageidentity/passage-node/lib/cjs/generated/models/Model401Error.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Model401ErrorToJSON = exports.Model401ErrorFromJSONTyped = exports.Model401ErrorFromJSON = exports.instanceOfModel401Error = exports.Model401ErrorCodeEnum = void 0;
/**
 * @export
 */
exports.Model401ErrorCodeEnum = {
    InvalidAccessToken: 'invalid_access_token'
};
/**
 * Check if a given object implements the Model401Error interface.
 */
function instanceOfModel401Error(value) {
    let isInstance = true;
    isInstance = isInstance && "code" in value;
    isInstance = isInstance && "error" in value;
    return isInstance;
}
exports.instanceOfModel401Error = instanceOfModel401Error;
function Model401ErrorFromJSON(json) {
    return Model401ErrorFromJSONTyped(json, false);
}
exports.Model401ErrorFromJSON = Model401ErrorFromJSON;
function Model401ErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': json['code'],
        'error': json['error'],
    };
}
exports.Model401ErrorFromJSONTyped = Model401ErrorFromJSONTyped;
function Model401ErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'error': value.error,
    };
}
exports.Model401ErrorToJSON = Model401ErrorToJSON;
