// @passageidentity/passage-node/lib/cjs/generated/models/Technologies.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TechnologiesToJSON = exports.TechnologiesFromJSONTyped = exports.TechnologiesFromJSON = exports.Technologies = void 0;
/**
 *
 * @export
 */
exports.Technologies = {
    React: 'react',
    Go: 'go',
    Vue: 'vue',
    Angular: 'angular',
    Python: 'python',
    Javascript: 'javascript',
    Ios: 'ios',
    Android: 'android'
};
function TechnologiesFromJSON(json) {
    return TechnologiesFromJSONTyped(json, false);
}
exports.TechnologiesFromJSON = TechnologiesFromJSON;
function TechnologiesFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.TechnologiesFromJSONTyped = TechnologiesFromJSONTyped;
function TechnologiesToJSON(value) {
    return value;
}
exports.TechnologiesToJSON = TechnologiesToJSON;
