// @passageidentity/passage-node/lib/cjs/generated/models/UpdateUserRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateUserRequestToJSON = exports.UpdateUserRequestFromJSONTyped = exports.UpdateUserRequestFromJSON = exports.instanceOfUpdateUserRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UpdateUserRequest interface.
 */
function instanceOfUpdateUserRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateUserRequest = instanceOfUpdateUserRequest;
function UpdateUserRequestFromJSON(json) {
    return UpdateUserRequestFromJSONTyped(json, false);
}
exports.UpdateUserRequestFromJSON = UpdateUserRequestFromJSON;
function UpdateUserRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'phone': !(0, runtime_1.exists)(json, 'phone') ? undefined : json['phone'],
        'user_metadata': !(0, runtime_1.exists)(json, 'user_metadata') ? undefined : json['user_metadata'],
    };
}
exports.UpdateUserRequestFromJSONTyped = UpdateUserRequestFromJSONTyped;
function UpdateUserRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
        'phone': value.phone,
        'user_metadata': value.user_metadata,
    };
}
exports.UpdateUserRequestToJSON = UpdateUserRequestToJSON;
