// @passageidentity/passage-node/lib/cjs/generated/models/UserMetadataFieldType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserMetadataFieldTypeToJSON = exports.UserMetadataFieldTypeFromJSONTyped = exports.UserMetadataFieldTypeFromJSON = exports.UserMetadataFieldType = void 0;
/**
 *
 * @export
 */
exports.UserMetadataFieldType = {
    String: 'string',
    Boolean: 'boolean',
    Integer: 'integer',
    Date: 'date',
    Phone: 'phone',
    Email: 'email'
};
function UserMetadataFieldTypeFromJSON(json) {
    return UserMetadataFieldTypeFromJSONTyped(json, false);
}
exports.UserMetadataFieldTypeFromJSON = UserMetadataFieldTypeFromJSON;
function UserMetadataFieldTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UserMetadataFieldTypeFromJSONTyped = UserMetadataFieldTypeFromJSONTyped;
function UserMetadataFieldTypeToJSON(value) {
    return value;
}
exports.UserMetadataFieldTypeToJSON = UserMetadataFieldTypeToJSON;
