// @passageidentity/passage-node/lib/cjs/generated/models/UserResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResponseToJSON = exports.UserResponseFromJSONTyped = exports.UserResponseFromJSON = exports.instanceOfUserResponse = void 0;
const UserInfo_1 = require("./UserInfo");
/**
 * Check if a given object implements the UserResponse interface.
 */
function instanceOfUserResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfUserResponse = instanceOfUserResponse;
function UserResponseFromJSON(json) {
    return UserResponseFromJSONTyped(json, false);
}
exports.UserResponseFromJSON = UserResponseFromJSON;
function UserResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': (0, UserInfo_1.UserInfoFromJSON)(json['user']),
    };
}
exports.UserResponseFromJSONTyped = UserResponseFromJSONTyped;
function UserResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': (0, UserInfo_1.UserInfoToJSON)(value.user),
    };
}
exports.UserResponseToJSON = UserResponseToJSON;
