// @passageidentity/passage-node/lib/cjs/generated/models/UserMetadataField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserMetadataFieldToJSON = exports.UserMetadataFieldFromJSONTyped = exports.UserMetadataFieldFromJSON = exports.instanceOfUserMetadataField = void 0;
const UserMetadataFieldType_1 = require("./UserMetadataFieldType");
/**
 * Check if a given object implements the UserMetadataField interface.
 */
function instanceOfUserMetadataField(value) {
    let isInstance = true;
    isInstance = isInstance && "field_name" in value;
    isInstance = isInstance && "friendly_name" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "profile" in value;
    isInstance = isInstance && "registration" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfUserMetadataField = instanceOfUserMetadataField;
function UserMetadataFieldFromJSON(json) {
    return UserMetadataFieldFromJSONTyped(json, false);
}
exports.UserMetadataFieldFromJSON = UserMetadataFieldFromJSON;
function UserMetadataFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'field_name': json['field_name'],
        'friendly_name': json['friendly_name'],
        'id': json['id'],
        'profile': json['profile'],
        'registration': json['registration'],
        'type': (0, UserMetadataFieldType_1.UserMetadataFieldTypeFromJSON)(json['type']),
    };
}
exports.UserMetadataFieldFromJSONTyped = UserMetadataFieldFromJSONTyped;
function UserMetadataFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'field_name': value.field_name,
        'friendly_name': value.friendly_name,
        'id': value.id,
        'profile': value.profile,
        'registration': value.registration,
        'type': (0, UserMetadataFieldType_1.UserMetadataFieldTypeToJSON)(value.type),
    };
}
exports.UserMetadataFieldToJSON = UserMetadataFieldToJSON;
