// @passageidentity/passage-node/lib/cjs/generated/models/WebAuthnType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAuthnTypeToJSON = exports.WebAuthnTypeFromJSONTyped = exports.WebAuthnTypeFromJSON = exports.WebAuthnType = void 0;
/**
 * The type of this credential
 * @export
 */
exports.WebAuthnType = {
    Passkey: 'passkey',
    SecurityKey: 'security_key',
    Platform: 'platform'
};
function WebAuthnTypeFromJSON(json) {
    return WebAuthnTypeFromJSONTyped(json, false);
}
exports.WebAuthnTypeFromJSON = WebAuthnTypeFromJSON;
function WebAuthnTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.WebAuthnTypeFromJSONTyped = WebAuthnTypeFromJSONTyped;
function WebAuthnTypeToJSON(value) {
    return value;
}
exports.WebAuthnTypeToJSON = WebAuthnTypeToJSON;
