// @passageidentity/passage-node/lib/esm/generated/models/AppResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { AppInfoFromJSON, AppInfoToJSON, } from './AppInfo';
/**
 * Check if a given object implements the AppResponse interface.
 */
export function instanceOfAppResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "app" in value;
    return isInstance;
}
export function AppResponseFromJSON(json) {
    return AppResponseFromJSONTyped(json, false);
}
export function AppResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'app': AppInfoFromJSON(json['app']),
    };
}
export function AppResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'app': AppInfoToJSON(value.app),
    };
}
