// @passageidentity/passage-node/lib/esm/generated/models/ElementCustomization.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FontFamilyFromJSON, FontFamilyToJSON, } from './FontFamily';
/**
 * Check if a given object implements the ElementCustomization interface.
 */
export function instanceOfElementCustomization(value) {
    let isInstance = true;
    return isInstance;
}
export function ElementCustomizationFromJSON(json) {
    return ElementCustomizationFromJSONTyped(json, false);
}
export function ElementCustomizationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'passage_container_background_color': !exists(json, 'passage_container_background_color') ? undefined : json['passage_container_background_color'],
        'passage_container_max_width': !exists(json, 'passage_container_max_width') ? undefined : json['passage_container_max_width'],
        'passage_input_box_background_color': !exists(json, 'passage_input_box_background_color') ? undefined : json['passage_input_box_background_color'],
        'passage_input_box_border_radius': !exists(json, 'passage_input_box_border_radius') ? undefined : json['passage_input_box_border_radius'],
        'passage_header_font_family': !exists(json, 'passage_header_font_family') ? undefined : FontFamilyFromJSON(json['passage_header_font_family']),
        'passage_body_font_family': !exists(json, 'passage_body_font_family') ? undefined : FontFamilyFromJSON(json['passage_body_font_family']),
        'passage_header_text_color': !exists(json, 'passage_header_text_color') ? undefined : json['passage_header_text_color'],
        'passage_body_text_color': !exists(json, 'passage_body_text_color') ? undefined : json['passage_body_text_color'],
        'passage_primary_button_background_color': !exists(json, 'passage_primary_button_background_color') ? undefined : json['passage_primary_button_background_color'],
        'passage_primary_button_text_color': !exists(json, 'passage_primary_button_text_color') ? undefined : json['passage_primary_button_text_color'],
        'passage_primary_button_hover_color': !exists(json, 'passage_primary_button_hover_color') ? undefined : json['passage_primary_button_hover_color'],
        'passage_primary_button_border_radius': !exists(json, 'passage_primary_button_border_radius') ? undefined : json['passage_primary_button_border_radius'],
        'passage_primary_button_border_color': !exists(json, 'passage_primary_button_border_color') ? undefined : json['passage_primary_button_border_color'],
        'passage_primary_button_border_width': !exists(json, 'passage_primary_button_border_width') ? undefined : json['passage_primary_button_border_width'],
        'passage_secondary_button_background_color': !exists(json, 'passage_secondary_button_background_color') ? undefined : json['passage_secondary_button_background_color'],
        'passage_secondary_button_text_color': !exists(json, 'passage_secondary_button_text_color') ? undefined : json['passage_secondary_button_text_color'],
        'passage_secondary_button_hover_color': !exists(json, 'passage_secondary_button_hover_color') ? undefined : json['passage_secondary_button_hover_color'],
        'passage_secondary_button_border_radius': !exists(json, 'passage_secondary_button_border_radius') ? undefined : json['passage_secondary_button_border_radius'],
        'passage_secondary_button_border_color': !exists(json, 'passage_secondary_button_border_color') ? undefined : json['passage_secondary_button_border_color'],
        'passage_secondary_button_border_width': !exists(json, 'passage_secondary_button_border_width') ? undefined : json['passage_secondary_button_border_width'],
    };
}
export function ElementCustomizationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'passage_container_background_color': value.passage_container_background_color,
        'passage_container_max_width': value.passage_container_max_width,
        'passage_input_box_background_color': value.passage_input_box_background_color,
        'passage_input_box_border_radius': value.passage_input_box_border_radius,
        'passage_header_font_family': FontFamilyToJSON(value.passage_header_font_family),
        'passage_body_font_family': FontFamilyToJSON(value.passage_body_font_family),
        'passage_header_text_color': value.passage_header_text_color,
        'passage_body_text_color': value.passage_body_text_color,
        'passage_primary_button_background_color': value.passage_primary_button_background_color,
        'passage_primary_button_text_color': value.passage_primary_button_text_color,
        'passage_primary_button_hover_color': value.passage_primary_button_hover_color,
        'passage_primary_button_border_radius': value.passage_primary_button_border_radius,
        'passage_primary_button_border_color': value.passage_primary_button_border_color,
        'passage_primary_button_border_width': value.passage_primary_button_border_width,
        'passage_secondary_button_background_color': value.passage_secondary_button_background_color,
        'passage_secondary_button_text_color': value.passage_secondary_button_text_color,
        'passage_secondary_button_hover_color': value.passage_secondary_button_hover_color,
        'passage_secondary_button_border_radius': value.passage_secondary_button_border_radius,
        'passage_secondary_button_border_color': value.passage_secondary_button_border_color,
        'passage_secondary_button_border_width': value.passage_secondary_button_border_width,
    };
}
