// @passageidentity/passage-node/lib/esm/generated/models/LayoutConfig.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the LayoutConfig interface.
 */
export function instanceOfLayoutConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "h" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "w" in value;
    isInstance = isInstance && "x" in value;
    isInstance = isInstance && "y" in value;
    return isInstance;
}
export function LayoutConfigFromJSON(json) {
    return LayoutConfigFromJSONTyped(json, false);
}
export function LayoutConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'h': json['h'],
        'id': json['id'],
        'w': json['w'],
        'x': json['x'],
        'y': json['y'],
    };
}
export function LayoutConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'h': value.h,
        'id': value.id,
        'w': value.w,
        'x': value.x,
        'y': value.y,
    };
}
