// @passageidentity/passage-node/lib/esm/generated/models/Layouts.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { LayoutConfigFromJSON, LayoutConfigToJSON, } from './LayoutConfig';
/**
 * Check if a given object implements the Layouts interface.
 */
export function instanceOfLayouts(value) {
    let isInstance = true;
    isInstance = isInstance && "profile" in value;
    isInstance = isInstance && "registration" in value;
    return isInstance;
}
export function LayoutsFromJSON(json) {
    return LayoutsFromJSONTyped(json, false);
}
export function LayoutsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'profile': (json['profile'].map(LayoutConfigFromJSON)),
        'registration': (json['registration'].map(LayoutConfigFromJSON)),
    };
}
export function LayoutsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'profile': (value.profile.map(LayoutConfigToJSON)),
        'registration': (value.registration.map(LayoutConfigToJSON)),
    };
}
