// @passageidentity/passage-node/lib/esm/generated/models/ListDevicesResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { WebAuthnDevicesFromJSON, WebAuthnDevicesToJSON, } from './WebAuthnDevices';
/**
 * Check if a given object implements the ListDevicesResponse interface.
 */
export function instanceOfListDevicesResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "devices" in value;
    return isInstance;
}
export function ListDevicesResponseFromJSON(json) {
    return ListDevicesResponseFromJSONTyped(json, false);
}
export function ListDevicesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'devices': (json['devices'].map(WebAuthnDevicesFromJSON)),
    };
}
export function ListDevicesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'devices': (value.devices.map(WebAuthnDevicesToJSON)),
    };
}
