// @passageidentity/passage-node/lib/esm/generated/models/MagicLink.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { MagicLinkTypeFromJSON, MagicLinkTypeToJSON, } from './MagicLinkType';
/**
 * Check if a given object implements the MagicLink interface.
 */
export function instanceOfMagicLink(value) {
    let isInstance = true;
    isInstance = isInstance && "activated" in value;
    isInstance = isInstance && "app_id" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "identifier" in value;
    isInstance = isInstance && "redirect_url" in value;
    isInstance = isInstance && "secret" in value;
    isInstance = isInstance && "ttl" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "user_id" in value;
    return isInstance;
}
export function MagicLinkFromJSON(json) {
    return MagicLinkFromJSONTyped(json, false);
}
export function MagicLinkFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'activated': json['activated'],
        'app_id': json['app_id'],
        'id': json['id'],
        'identifier': json['identifier'],
        'redirect_url': json['redirect_url'],
        'secret': json['secret'],
        'ttl': json['ttl'],
        'type': MagicLinkTypeFromJSON(json['type']),
        'url': json['url'],
        'user_id': json['user_id'],
    };
}
export function MagicLinkToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'activated': value.activated,
        'app_id': value.app_id,
        'id': value.id,
        'identifier': value.identifier,
        'redirect_url': value.redirect_url,
        'secret': value.secret,
        'ttl': value.ttl,
        'type': MagicLinkTypeToJSON(value.type),
        'url': value.url,
        'user_id': value.user_id,
    };
}
