// @passageidentity/passage-node/lib/esm/generated/models/Model400Error.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * @export
 */
export const Model400ErrorCodeEnum = {
    InvalidRequest: 'invalid_request',
    CustomProviderRequired: 'custom_provider_required'
};
/**
 * Check if a given object implements the Model400Error interface.
 */
export function instanceOfModel400Error(value) {
    let isInstance = true;
    isInstance = isInstance && "code" in value;
    isInstance = isInstance && "error" in value;
    return isInstance;
}
export function Model400ErrorFromJSON(json) {
    return Model400ErrorFromJSONTyped(json, false);
}
export function Model400ErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': json['code'],
        'error': json['error'],
    };
}
export function Model400ErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'error': value.error,
    };
}
