// @passageidentity/passage-node/lib/esm/generated/models/UserResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { UserInfoFromJSON, UserInfoToJSON, } from './UserInfo';
/**
 * Check if a given object implements the UserResponse interface.
 */
export function instanceOfUserResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
export function UserResponseFromJSON(json) {
    return UserResponseFromJSONTyped(json, false);
}
export function UserResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': UserInfoFromJSON(json['user']),
    };
}
export function UserResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': UserInfoToJSON(value.user),
    };
}
