// @passageidentity/passage-node/lib/esm/generated/models/UserEventInfo.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the UserEventInfo interface.
 */
export function instanceOfUserEventInfo(value) {
    let isInstance = true;
    isInstance = isInstance && "created_at" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "ip_addr" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "user_agent" in value;
    return isInstance;
}
export function UserEventInfoFromJSON(json) {
    return UserEventInfoFromJSONTyped(json, false);
}
export function UserEventInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'created_at': (new Date(json['created_at'])),
        'id': json['id'],
        'ip_addr': json['ip_addr'],
        'type': json['type'],
        'user_agent': json['user_agent'],
    };
}
export function UserEventInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'created_at': (value.created_at.toISOString()),
        'id': value.id,
        'ip_addr': value.ip_addr,
        'type': value.type,
        'user_agent': value.user_agent,
    };
}
