// @passageidentity/passage-node/lib/esm/generated/models/WebAuthnDevices.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: v1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { WebAuthnTypeFromJSON, WebAuthnTypeToJSON, } from './WebAuthnType';
/**
 * Check if a given object implements the WebAuthnDevices interface.
 */
export function instanceOfWebAuthnDevices(value) {
    let isInstance = true;
    isInstance = isInstance && "created_at" in value;
    isInstance = isInstance && "cred_id" in value;
    isInstance = isInstance && "friendly_name" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "last_login_at" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "updated_at" in value;
    isInstance = isInstance && "usage_count" in value;
    return isInstance;
}
export function WebAuthnDevicesFromJSON(json) {
    return WebAuthnDevicesFromJSONTyped(json, false);
}
export function WebAuthnDevicesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'created_at': (new Date(json['created_at'])),
        'cred_id': json['cred_id'],
        'friendly_name': json['friendly_name'],
        'id': json['id'],
        'last_login_at': (new Date(json['last_login_at'])),
        'type': WebAuthnTypeFromJSON(json['type']),
        'updated_at': (new Date(json['updated_at'])),
        'usage_count': json['usage_count'],
    };
}
export function WebAuthnDevicesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'created_at': (value.created_at.toISOString()),
        'cred_id': value.cred_id,
        'friendly_name': value.friendly_name,
        'id': value.id,
        'last_login_at': (value.last_login_at.toISOString()),
        'type': WebAuthnTypeToJSON(value.type),
        'updated_at': (value.updated_at.toISOString()),
        'usage_count': value.usage_count,
    };
}
